### PR TITLE
Show a sample editor on the CodeMirror plugin config page.

### PR DESCRIPTION
**Pull request by okonomiyaki3000**
Pull Request for Issue # .

### Summary of Changes
Added a CodeMirror editor to the 'Appearance Options' tab of the CodeMirror plugin config page to take the guesswork out of configuring CodeMirror's appearance. 

### Testing Instructions
Open the CodeMirror config page and click on th 'Appearance Options' tab. Change the theme, font, etc. Save changes. 

### Expected result
The new appearance settings will be shown in the embedded editor field.

### Actual result
Should be as expected.

### Documentation Changes Required
No but there are a couple of new strings in the language file.

### CODE REVIEW

**Comment by ghost** [comment]
I have tested this item :white_check_mark: successfully on f2fac83e498b8782a2f9d9ecce4c2394881f9eb1<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/17042">issues.joomla.org/tracker/joomla-cms/17042</a>.</sub>

**Comment by brianteeman** [comment]
There must be a better way of doing this without adding inline jquery javascript and it really needs to be something that is translatable and not hard coded in english

**Comment by okonomiyaki3000** [comment]
@brianteeman 🤣  I think you've missed something. The "inline jquery" is just sample js code. It doesn't execute, it just appears in the editor so you see what it will look like. The only hard coded English is "Hello World" which I think is probably acceptable but, if not, I can change it.

**Comment by brianteeman** [comment]
Serves me right for replying before coffee

I understand what you are doing now - not sure of the benefit of having it - when I looked at the options page it confused me more than it helped me

Maybe it needs a different label than "sample text"

Layout preview?
Configuration preview?

**Comment by okonomiyaki3000** [comment]
Maybe just 'Preview'?

**Comment by okonomiyaki3000** [comment]
Here's that `pr-merge` error again. What is this?

**Comment by okonomiyaki3000** [comment]
Don't believe that `pr-merge` error. There is no problem with this PR.

File: plugins/editors/codemirror/codemirror.xml [hunk 1]
```diff
@@ -320,6 +320,54 @@
 					<option value="simple">PLG_CODEMIRROR_FIELD_VALUE_SCROLLBARSTYLE_SIMPLE</option>
 					<option value="overlay">PLG_CODEMIRROR_FIELD_VALUE_SCROLLBARSTYLE_OVERLAY</option>
 				</field>
+
+				<field
+					name="preview"
+					type="editor"
+					editor="codemirror"
```
Comment (Quy) [inline plugins/editors/codemirror/codemirror.xml]: The four attributes `name`, `type`, `label` and `description` should be written in this order per https://developer.joomla.org/coding-standards/xml.html

Comment (okonomiyaki3000) [reply plugins/editors/codemirror/codemirror.xml]: Thanks! Didn't know we had that for xml.

File: administrator/language/en-GB/en-GB.plg_editors_codemirror.ini [hunk 2]
```diff
@@ -37,6 +37,8 @@ PLG_CODEMIRROR_FIELD_MATCHBRACKETS_DESC="Highlight matching brackets."
 PLG_CODEMIRROR_FIELD_MATCHBRACKETS_LABEL="Match Brackets"
 PLG_CODEMIRROR_FIELD_MATCHTAGS_DESC="Highlight matching tags."
 PLG_CODEMIRROR_FIELD_MATCHTAGS_LABEL="Match Tags"
+PLG_CODEMIRROR_FIELD_PREVIEW_DESC="An example of what your CodeMirror editor fields will look like with the current settings (save to update)."
```
Comment (Quy) [inline administrator/language/en-GB/en-GB.plg_editors_codemirror.ini:40]: How about this? `A preview of the CodeMirror editor with the current settings (save to update).`

File: plugins/editors/codemirror/codemirror.xml [hunk 3]
```diff
@@ -320,6 +320,54 @@
...
+
+				<field
+					name="preview"
+					type="editor"
+					label="PLG_CODEMIRROR_FIELD_PREVIEW_LABEL"
+					description="PLG_CODEMIRROR_FIELD_PREVIEW_DESC"
+					editor="codemirror"
+					filter="unset"
+					buttons="false"
+				>
```
Comment (Quy) [inline plugins/editors/codemirror/codemirror.xml]: Indent.

**Comment by Quy** [comment]
It would be nice to see the preview when Theme is changed without having to click Save button each time.

**Comment by brianteeman** [comment]
I have tested this item :white_check_mark: successfully on 2d6a9d5d59f5fe9813d0480bab3e4904623f3142<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/17042">issues.joomla.org/tracker/joomla-cms/17042</a>.</sub>

**Comment by carlitorweb** [comment]
I have tested this item :white_check_mark: successfully on 2d6a9d5d59f5fe9813d0480bab3e4904623f3142<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/17042">issues.joomla.org/tracker/joomla-cms/17042</a>.</sub>

**Comment by Quy** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/17042">issues.joomla.org/tracker/joomla-cms/17042</a>.</sub>

**Pull request merged**